docs/hoon/library/2k.md

```diff
@@ -180,164 +174,192 @@ Source
 
-    > =s (~(gas to *(qeu @)) `(list @)`~[1 2 3])
+#### Examples
+
+    > =s (~(gas to *(tree @)) `(list @)`~[1 2 3])
     > ~(get to s)
-    [p=1 q={3 2}]
+    [n=2 l={3} r={1}]
+
     > ~(get to ~)
     ! exit
 
-------------------------------------------------------------------------
-
+***
 ### `+-nap:to`
 
-Remove last element 
+Remove head
 
-Removes the head of `++queue` `a`, producing the resulting queue.
+Removes the head of queue `a`, producing the resulting queue.
 
-Accepts
--------
+#### Accepts
 
 `a` is a queue.
 
-Produces
---------
+#### Produces
 
 A queue.
 
-Source
-------
+#### Source
 
-      +-  nap                                               ::  removes head
+      +-  nap                                              
         ?>  ?=(^ a)
-        ?:  =(~ l.a)  r.a
+        ?~  l.a  r.a
         =+  b=get(a l.a)
         bal(a ^+(a [p.b q.b r.a]))
-      ::
 
-Examples
---------
+#### Examples
 
     > =a (~(gas to `(qeu @)`~) `(list @)`[1 2 3 4 5 6 ~])
     > -.a
     n=6
+
     > =b ~(nap to a)
     > -.b
     n=2
+
+    > a
+    [n=6 l={} r={5 4 3 2 1}]
+
     > b
+    [n=2 l=[n=4 l={5} r={3}] r=[n=1 l={} r={}]]
+
+    > `(qeu)`b
     {5 4 3 2 1}
-    > a
+
+    > `(qeu)`a
     {6 5 4 3 2 1}
 
-------------------------------------------------------------------------
+***
+###  `+-nip:to`
+
+Removes the root of queue `a`, producing the resulting queue.
+
+#### Accepts
+
+`a` is a queue.
+
+#### Produces
+
+A queue.
+
+#### Source
+
+      +-  nip                                               
+        |-  ^+  a
+        ?~  a  ~
+        ?~  l.a  r.a
+        ?~  r.a  l.a
+        ?:  (vor n.l.a n.r.a)
+          [n.l.a l.l.a $(l.a r.l.a)]
+        [n.r.a $(r.a l.r.a) r.r.a]
+
+#### Examples
+
+    > =a (~(gas to `(qeu @)`~) `(list @)`[1 2 3 4 5 6 ~])
+    > a
+    [n=6 l={} r={5 4 3 2 1}]
 
+    > ~(nip to a)
+    [n=2 l=[n=4 l={5} r={3}] r=[n=1 l={} r={}]]
+
+***
 ### `+-put:to`
 
 Insert
 
-Accept any noun `b` and adds to `++queue` `a` as the head, producing the
+Accept any noun `b` and adds to queue `a` as the head, producing the
 resulting queue.
 
-Accepts
--------
+#### Accepts
 
 `a` is a queue.
 
 `b` is any noun.
 
-Produces
---------
+#### Produces
 
 A queue.
 
-Source
-------
+#### Source
 
-      +-  put                                               ::  insert new tail
-        |*  b/*
+      +-  put                                            
+        |*  b=*
         |-  ^+  a
         ?~  a
           [b ~ ~]
         bal(a a(l $(a l.a)))
-      ::
 
+#### Examples
 
-Examples
---------
-
-    > (~(gas to `(qeu @)`~) `(list @)`[3 1 2 4 5 6 ~])
-    > (~(put to a) 7)
+    > =a (~(gas to `(qeu @)`~) `(list @)`[3 1 2 4 5 6 ~])
+    > `(qeu)`(~(put to a) 7)
     {7 6 5 4 2 1 3}
 
-------------------------------------------------------------------------
-
+***
 ### `+-tap:to`
 
 Queue to list
 
-Produces `++queue` `a` as a `++list` from front to back.
+Produces queue `a` as a list from front to back.
```

Comment:
Does this mean it produces the elements in FIFO order, or the reverse? the "front" and "back" concepts don't seem totally clear to me.